setup.py

```diff
@@ -61,6 +64,12 @@
     )
     ext_modules.append(ext)
 
+# Hack to install ta-lib library loading into virtualenv
+def install_virtualenv_lib_loader():
+    with open(os.getenv('VIRTUAL_ENV', './venv') + '/bin/activate', 'a') as f:
```

Comment:
Yes, you are correct. Such a check is necessary.

When you create a virtualenv the following standard directory structure appears (`bin`, `include`, `lib`, `local`, `src`) and Python binaries and packages get installed into it thereafter.  There is no standard location where you create a `$VIRTUAL_ENV`, some put it in a subdirectory of their project `project/venv` others directly in project `project`. Nevertheless this works great for all Python-related things, you just run `. venv/bin/activate` and use `pip install` and other things as usual, knowing it will install everything inside the virtualenv. As far as I know the best way to detect a virtualenv is using `$VIRTUAL_ENV` environment variable.

Unfortunately non-Python libraries are not handled by virtualenv at all. Nevertheless in my setups I usually install them inside the virtualenv directory (`./configure --prefix=$VIRTUAL_ENV`). This way binaries get installed into `./venv/bin` and work automatically (because `$PATH` is configured correctly by ./venv/bin/activate`) and libraries into `./venv/lib`. After adding a line such as `export LD_LIBRARY_PATH=$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH` they also work as expected.

In case this seems to you as an ugly workaround, you may want to add an option to specify a custom `includes` and `lib` folder to search for the TA-Lib (eg. through an environment variable).